### DISCUSSION TITLE
Adding a language/localization plugin

### DISCUSSION THREAD

**Discussion by mrzdcmps** [jspsych/jsPsych#3311] [Ideas]
It would be great if a language selection-plugin could be added.
Ideally, this would set the relevant buttons-texts automatically for all following trials.
Dynamic text, as in instructions or stimuli could be set by specifying an object accordingly. Individual trials would need to be uniquely identified, e.g. by trial-name, via a "data"-attribute or by going through the timeline one by one.

```
const language = {
  type: jsPsychLanguage,
  languages: ["en", "de"],
  automaticDetection: false,
  translations: {
      "en": {
          "welcome": {
               "stimulus": "Welcome to the experiment. Press any key to begin."
           },
          "instructions:": {
               "stimulus": "<strong>Press the F key</strong>"
           }
      },
      "de": {
          "welcome": {
               "stimulus": "Willkommen zum Experiment. Drücken Sie eine beliebige Taste, um zu beginnen."
           },
          "instructions:": {
                "stimulus": "<strong>Drücken Sie die F-Taste</strong>"
           }
      }
  }
}
```

**Comment by Shaobin-Jiang** [comment]
I personally do not think this is a good idea. One cool fact about jsPsych is that its trials are independent of one another, and the addition of your proposed plugin breaks that. If we really need i18n related config I believe it should be made configurable in `initJsPsych`.

**Reply by mrzdcmps** [reply]
Yes, you are right. It makes more sense to address the translations in the `initJsPsych`.

**Comment by jodeleeuw** [comment]
There's a work in progress proposal for doing this in https://github.com/jspsych/jsPsych/pull/3107. Would welcome feedback there!
